UIKit_DZ/UIKit_DZ/AuthScreen/AuthViewController.swift

```diff
@@ -3,5 +3,171 @@
+        divider.backgroundColor = .gray
+        divider.frame = CGRect(x: 16, y: lineY, width: 335, height: 1)
+        return divider
+    }
+
+    /// создание textFields
+    private func createTextFields(
+        textField: UITextField, placeholder: String, lineY: Int, isSecure: Bool = false
+    ) {
+        textField.placeholder = placeholder
+        textField.borderStyle = .none
+        textField.isSecureTextEntry = isSecure
+        textField.frame = CGRect(x: 16, y: lineY, width: 175, height: 20)
+        textField.addTarget(self, action: #selector(textFieldDidEndEditing), for: .editingChanged)
+        whiteView.addSubview(textField)
+    }
+
+    /// метод для кнопки показывать или сскрывать пароль
+    private func createShowPasswordButton() {
+        showPasswordButton.setImage(UIImage(systemName: "eye.slash.fill"), for: .normal)
+        showPasswordButton.frame = CGRect(x: 332, y: 185, width: 22, height: 19)
+        showPasswordButton.addTarget(self, action: #selector(showPassword), for: .touchUpInside)
+        showPasswordButton.tintColor = .gray
+        whiteView.addSubview(showPasswordButton)
+    }
+
+    /// кнопка присоединиться
+    private func createSignUpButton() {
+        sigUpButton.center.x = view.center.x
+        sigUpButton.backgroundColor = #colorLiteral(red: 0.80188936, green: 0.9210196137, blue: 0.9322513938, alpha: 1)
+        sigUpButton.setTitle("Войти", for: .normal)
+        sigUpButton.isEnabled = false
+        sigUpButton.addTarget(self, action: #selector(presentSecondVC), for: .touchUpInside)
+        view.addSubview(sigUpButton)
+    }
+
+    /// кнопка показать/скрыть пароль
+    @objc private func showPassword() {
+        passwordTextField.isSecureTextEntry.toggle()
+        showPasswordButton.setImage(
+            UIImage(systemName: !passwordTextField.isSecureTextEntry ? "eye.slash.fill" : "eye.fill"),
+            for: .normal
+        )
+    }
+
+    /// пушим следующий экран
+    @objc private func presentSecondVC() {
+        let menuVC = MenuViewController()
+        let navigationVC = UINavigationController(rootViewController: menuVC)
+        navigationVC.modalPresentationStyle = .fullScreen
+        present(navigationVC, animated: true)
+    }
+}
+
+/// Расширение
+extension AuthViewController: UITextFieldDelegate {
+    @objc func textFieldDidEndEditing(_ textField: UITextField) {
+        let textPassword = passwordTextField.text ?? ""
+        let textLogIn = loginTextField.text ?? ""
+        if !textPassword.isEmpty, !textLogIn.isEmpty {
+            sigUpButton.backgroundColor = #colorLiteral(red: 0.3460897207, green: 0.7455198169, blue: 0.7799417377, alpha: 1)
+            sigUpButton.isEnabled = true
+        } else {
+            sigUpButton.backgroundColor = #colorLiteral(red: 0.80188936, green: 0.9210196137, blue: 0.9322513938, alpha: 1)
+            sigUpButton.isEnabled = false
+        }
+    }
+}
+
+final class UniversalButton: UIButton {
```

Comment:
```suggestion
final class UniversalButton: UIButton {
```
Universal Custom Model это все стоп слова